spam/spam_tutorial.py

```diff
@@ -625,61 +597,127 @@ def plot_probabilities_histogram(Y_probs):
 X_test = vectorizer.transform(words_test)
 
 # %% [markdown]
-# * Filter out examples with no labels (always abstains)
+# ### Filtering Unlabeled Data Points
+
+# %% [markdown]
+# As we saw earlier, some of the data points in our training set received no weak labels from our LFs.
+# These examples are not helpful for training our classifier, as they convey no supervision signal, so we filter them out before training.
 
 # %%
 mask = L_train.sum(axis=1) != ABSTAIN * len(lfs)
 X_train = X_train[mask, :]
-Y_train = Y_train[mask]
+Y_probs_train = Y_probs_train[mask]
 
 # %% [markdown]
-# ### Keras Classifier
+# ### Keras Classifier with Probabilistic Labels
 
 # %% [markdown]
-# * TBD
+# Our Keras classifier is a simple logistic regression classifier.
+# We compile it with a `categorical_crossentropy` loss so that it can handle probabilistic labels instead of integer labels.
+# We use the common settings of an `Adam` optimizer and early stopping (evaluating the model on the validation set after each epoch and reloading the weights from when it achived the best score).
 
 # %%
-# TODO:
-# import simple logistic regression classifier in Keras
-# train on noise-aware loss
-# score
-# bonus: peek at performance w/r/t using hard (int) labels i/o soft (float) labels
+from snorkel.analysis.utils import probs_to_preds, preds_to_probs
+from snorkel.analysis.metrics import metric_score
+import tensorflow as tf
+
+keras_model = tf.keras.Sequential()
+keras_model.add(
+    tf.keras.layers.Dense(
+        2,
+        input_dim=X_train.shape[1],
+        activation=tf.nn.softmax,
+        kernel_regularizer=tf.keras.regularizers.l2(0.01),
+    )
+)
+optimizer = tf.keras.optimizers.Adam(lr=0.001)
+keras_model.compile(
+    optimizer=optimizer, loss="categorical_crossentropy", metrics=["accuracy"]
```

Comment:
comment for `categorical_crossentropy`

Reply:
Also explained in the text above the cell.